File: python/speech-transcribe/token-auth.py
import http.client
import json
import mimetypes
import os
import uuid

# token configuration
client_id = "[[client_id]]"
client_secret = "[[client_secret]]"
scopes = "[[scopes]]"

# get the access_token from the token endpoint
token_data = {
    "client_id": client_id,
    "client_secret": client_secret,
    "grant_type": "client_credentials",
    "scope": scopes,
}

token_json_data = json.dumps(token_data)

token_host = "api.intelligent-api.com"
token_path = "/v1/token"

token_connection = http.client.HTTPSConnection(token_host)

token_headers = {"Content-Type": "application/json"}

token_connection.request(
    "POST", token_path, body=token_json_data, headers=token_headers
)
token_response = token_connection.getresponse()
token_response_data = token_response.read().decode("utf-8")

token_json_response = json.loads(token_response_data)
token = token_json_response.get("access_token")

# use the access_token as a bearer token
authorization = "Bearer " + token

# populate endpoint parameters
file_path = "[[filepath]]"
user_agent = "Intelligent API Sample Python Code"

# build multipart form body
boundary = uuid.uuid4().hex
body = []

# form part headers
body.append(f"--{boundary}".encode())
body.append(
    f'Content-Disposition: form-data; name="file"; filename="{os.path.basename(file_path)}"'.encode()
)
mime_type = mimetypes.guess_type(file_path)[0] or "application/octet-stream"
body.append(f"Content-Type: {mime_type}".encode())
body.append(b"")

# file content
with open(file_path, "rb") as file:
    body.append(file.read())

# closing boundary
body.append(f"--{boundary}--".encode())

multipart_body = b"\r\n".join(body)

# invoke the API endpoint
host = "api.intelligent-api.com"
path = "/v1/speech/transcribe"

connection = http.client.HTTPSConnection(host)

headers = {
    "Authorization": authorization,
    "Content-Type": f"multipart/form-data; boundary={boundary}",
    "User-Agent": user_agent,
}

connection.request("POST", path, body=multipart_body, headers=headers)
response = connection.getresponse()
response_data = response.read().decode("utf-8")

print("Status Code:", response.status)
print("Response:", response_data)

connection.close()

File: java/speech-expenses/TokenAuth.java
import java.io.ByteArrayOutputStream;
import java.io.File;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.nio.file.Files;

public class TokenAuth {
        public static void main(String args[]) {
                // token configuration
                String clientId = "[[client_id]]";
                String clientSecret = "[[client_secret]]";
                String scopes = "[[scopes]]";

                // get the access_token from the token endpoint
                String tokenPayload = "{ \"client_id\": \"" + clientId + "\", \"client_secret\": \"" + clientSecret
                                + "\", \"grant_type\": \"client_credentials\", \"scope\": \"" + scopes + "\"  }";
                String token = "";

                try {
                        HttpClient client = HttpClient.newHttpClient();
                        HttpRequest request = HttpRequest.newBuilder()
                                        .uri(URI.create("https://api.intelligent-api.com/v1/token"))
                                        .header("Content-Type", "application/json")
                                        .POST(HttpRequest.BodyPublishers.ofString(tokenPayload))
                                        .build();

                        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

                        String searchKey = "access_token";
                        String tokenResponse = response.body();
                        int keyIndex = tokenResponse.indexOf(searchKey);
                        int valueStart = tokenResponse.indexOf("\"", keyIndex + searchKey.length()) + 3;
                        int valueEnd = tokenResponse.indexOf("\"", valueStart);
                        token = tokenResponse.substring(valueStart, valueEnd);
                } catch (Exception e) {
                        e.printStackTrace();
                        return;
                }

                // use the access_token as a bearer token
                String authorization = "Bearer ".concat(token);

                // populate endpoint parameters
                String fullPathToFile = "[[filepath]]";
                String userAgent = "Intelligent API Sample Java Code";

                // invoke the API endpoint
                try {
                        File file = new File(fullPathToFile);
                        String boundary = "----FormBoundary" + System.currentTimeMillis();

                        // Build multipart body
                        ByteArrayOutputStream bodyStream = new ByteArrayOutputStream();
                        PrintWriter writer = new PrintWriter(new OutputStreamWriter(bodyStream, "UTF-8"), true);

                        writer.append("--").append(boundary).append("\r\n");
                        writer.append("Content-Disposition: form-data; name=\"file\"; filename=\"")
                                        .append(file.getName())
                                        .append("\"").append("\r\n");
                        writer.append("Content-Type: application/octet-stream").append("\r\n");
                        writer.append("\r\n");
                        writer.flush();

                        bodyStream.write(Files.readAllBytes(file.toPath()));

                        writer.append("\r\n");
                        writer.append("--").append(boundary).append("--").append("\r\n");
                        writer.flush();

                        HttpClient client = HttpClient.newHttpClient();
                        HttpRequest request = HttpRequest.newBuilder()
                                        .uri(URI.create("https://api.intelligent-api.com/v1/speech/expenses?classifications=Food%20%26%20Dining,Transport,Shopping,Utilities,Entertainment,Health,Education,Other"))
                                        .header("Authorization", authorization)
                                        .header("User-Agent", userAgent)
                                        .header("Content-Type", "multipart/form-data; boundary=" + boundary)
                                        .POST(HttpRequest.BodyPublishers.ofByteArray(bodyStream.toByteArray()))
                                        .build();

                        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

                        System.out.println("Response Code: " + response.statusCode());
                        System.out.println("Response Body: " + response.body());
                } catch (Exception e) {
                        e.printStackTrace();
                }
        }
}
